src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -462,16 +462,15 @@ public bool Equals(AnalyzerReference other)
         {
             if (other != null)
             {
-                return other.Display == this.Display &&
-                       other.FullPath == this.FullPath;
+                return other.FullPath == this.FullPath;
             }
 
             return base.Equals(other);
```

Comment:
>return base.Equals(other); [](start = 12, length = 26)

It doesn't look like there is a good reason to delegate to the base class here. Consider simply returning false. Also, consider adding an abstract override for Equals and GetHashCode into the base class to make sure each leaf class provides proper implementation.